MapboxDirections/OfflineDirections.swift

```diff
@@ -0,0 +1,139 @@
+import Foundation
+
+
+public typealias OfflineDownloaderCompletionHandler = (_ location: URL?,_ response: URLResponse?, _ error: Error?) -> Void
+public typealias OfflineDownloaderProgressHandler = (_ bytesWritten: Int64, _ totalBytesWritten: Int64, _ totalBytesExpectedToWrite: Int64) -> Void
+public typealias OfflineVersionsHandler = (_ version: [Version]?, _ error: Error?) -> Void
+
+struct AvailableVersionsResponse: Codable {
+    let availableVersions: [Version]
+}
+
+@objc(MBVersion)
+public class Version: NSObject, Codable {
```

Comment:
There would be somewhat less boilerplate if we made Version a struct instead. It would have to be defined in Objective-C in order to support both languages.

Reply:
Good point. The version should be represented as a String according to the api team.